### ISSUE TITLE
Make OpenImageIO optional

### ISSUE THREAD

**Issue by sweeneychris** [sweeneychris/TheiaSfM#116]
We should be able to separate out the OpenImageIO dependency and allow for Theia to compile with only the files which do not depend on the FloatImage class.

**Comment by charlescva** [comment]
So just to verify, OpenImageIO is only used in:
exif_reader : https://github.com/sweeneychris/TheiaSfM/blob/master/src/theia/sfm/exif_reader.cc#L130

And significantly throughout the Image class : https://github.com/sweeneychris/TheiaSfM/blob/master/src/theia/image/image.cc#L39

Thats it?

**Comment by sweeneychris** [comment]
In a few other locations I use the iterators from open image IO since it is faster to loop over all image pixels with the iterator than to simply index the pixels (e.g. in the image undistortion tool)

**Comment by cdcseacave** [comment]
For reading the EXIF info [easyexif](https://github.com/mayanklahiri/easyexif) library can be used (stand alone), or [TinyEXIF](https://github.com/cdcseacave/TinyEXIF) library, an extension supporting also parsing the XMP tags.

**Comment by sweeneychris** [comment]
I found EasyEXIF to contain bugs that caused lots of segfaults. I reported
some of the issues on GitHub but they were never solved.

**Comment by cdcseacave** [comment]
Try my lib then, it was completely rewritten. And if you still find bugs, I will fix them as soon as reported.

**Comment by sweeneychris** [comment]
There doesn't seem to be any advantage to doing so while Theia still uses
OpenImageIO.

**Comment by cdcseacave** [comment]
Indeed, all the above comments when for the case you want to actually implement this _issue_.

**Comment by sweeneychris** [comment]
Ah, sorry I lost track that this was on the issue since I'm responding via
email. The bigger hurdle is to find a decent image library. I will keep
these exif libs in mind!

**Comment by rcasey74** [comment]
Have you considered the ImageMagick library as an alternative to OpenImageIO?
https://imagemagick.org/script/index.php
It seems to be a pretty good all around utility for image work. Also, I've had more success building
it than OpenImageIO, which has a lot of open issues building reliably.
